src/main/java/com/epam/page/object/generator/validators/AbstractValidator.java

```diff
@@ -66,12 +75,21 @@ public void validate(ValidationContext validationContext) {
         searchRules.addAll(isValidateAllSearchRules ? validationContext.getAllSearchRules()
             : validationContext.getValidRules());
 
-        searchRules.forEach(searchRule -> {
-            validationContext
-                .addValidationResult(
-                    !isValid(searchRule, validationContext) ? new ValidationResult(false, this,
-                        searchRule) : new ValidationResult(true, this, searchRule));
-        });
+        searchRules.stream()
+            .filter(searchRule -> supportedSearchRuleTypes != null
```

Comment:
Why you using supportedSearchRuleTypes when filtering searchRules?

Reply:
We decide if this validator should validate SearchRule with such type

Reply:
I mean why you checking if it's not null in each iteration

Reply:
Done